File: src/main/java/com/checkers/exceptions/CellIsEmptyException.java
package com.checkers.exceptions;

import com.checkers.board.Cell;

public class CellIsEmptyException extends RuntimeException {

    private static final String MSG_TEMPLATE = "Cell %s doesn't have any piece";

    public CellIsEmptyException(Cell cell) {
        super(MSG_TEMPLATE.formatted(cell));
    }
}

File: src/main/kotlin/com/checkers/exception/CellIsEmptyException.kt
package com.checkers.exception

import com.checkers.board.Cell
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(HttpStatus.BAD_REQUEST)
class CellIsEmptyException(cell: Cell) : RuntimeException("Cell $cell doesn't have any piece")
